plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/MavenServerManager.java

```diff
@@ -216,7 +218,16 @@ public JavaParameters buildMavenServerParameters() {
 
     parameters.getClassPath().addAll(classPath);
 
-    parameters.getVmParameters().add("-Xmx512m");
+    String vmArguments =
+        MoreObjects.firstNonNull(
+            System.getenv("CHE_WORKSPACE_MAVEN_SERVER_JAVA_OPTIONS"),
```

Comment:
I think injecting those values from DI container would be preferable. I understand that because they may be unbound if env vars are not set, but we can overcome this with optional injection in Guice.